### PR TITLE
Today's fix and cleanup collection

### PR DESCRIPTION
**Pull request by AnHardt**
For details see the commit headers.

Pick what you want.

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -2907,14 +2907,8 @@ inline void gcode_G28() {
...
-
-          #if HAS_BED_PROBE
-            do_blocking_move_to_z(z_dest);
-          #else
-            line_to_z(z_dest);
-            stepper.synchronize();
-          #endif
-
+          line_to_z(z_dest);
+          stepper.synchronize();
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:2885]: I figured on saving one function call if not needed. I know it's a very minor thing. In the long run I'm sure we can continue to consolidate all these movement helpers.

**Comment by thinkyhead** [comment]
Everything seems to make good sense. Amazing how much can be removed from `quick_home_xy`, but it makes sense too! It's just a diagonal endstop move, after all. After quick-homing, it still continues to home X and Y separately.

I'm not sure of the point of these in `quick_home_xy` though…

``` cpp
current_position[X_AXIS] = current_position[Y_AXIS] = 0;
. . .
destination[X_AXIS] = destination[Y_AXIS] = 0;
```

…since we're sending coordinates to do moves from the current `planner.position`, and we can send them any old way, so long as at the end of the process we update `current_position`. (I guess that's one of those unwritten "rules of Marlin" that we need to include in dev docs.)

I notice also that if we take `quick_home_xy` as an atomic function, we're not setting the `destination[Z_AXIS]` which might be "anything." (Whatever state it was left in by the last thing that used it.) So, we could put the body of `quick_home_xy` back into `gcode_G28` so it's obvious that `destination[Z_AXIS]` was already set a moment ago.

The other thing I see is that the use of `destination` might be extraneous in some places. When we call `line_to_destination` it updates the planner's position from `destination`, leaving `current_position` unaltered. When we call `line_to_current_position` it has the same effect, only you update `current_position` first, before calling it. So it looks like we can replace some calls to `line_to_destination` with `line_to_current_position` in those cases where we're updating `current_position` from `destination` anyway.

**Comment by thinkyhead** [comment]
And… What was the point of this in `quick_home_xy` prior to now? It was sitting in there without any comment:

``` cpp
#if DISABLED(SCARA)
  current_position[Z_AXIS] = destination[Z_AXIS];
#endif
```

**Comment by AnHardt** [comment]
```
destination[X_AXIS] = destination[Y_AXIS] = 0;
```

With out this line i got grinding. Likely a later move relies on destination. 
I found this by try and error. Maybe after cleaning up endstop activation it's not needed anymore.
It's more or less a simplified echo of the former axis_is_at_home and destination = current.

Picking the right move is tricky.
For example in https://github.com/AnHardt/Marlin/pull/34/ i could save a lot of code by using `prepare_move_to_destination()` instead of reinventing it. But its a trade off size against time. `prepare_move_to_destination()` does additionally MBL splitting and clipping to software endstops, what is not a bad thing - except it is a bit slower than before.

**EDITED**

**Comment by AnHardt** [comment]
SCARA. I really don't know. Especially besause the z-axis in scara is not anything special. As long we do not get more then once a year a complain like "It does not work", i don't care much about scara. As long as we don't have at least one scara user who is willing to work with us it will not make sense to care about it. It will be shooting in the dark.

However - Z is not, and was not, involve in this moves.

Anyone who wants to send me a SCARA-machine?

**EDITED**

**Comment by thinkyhead** [comment]
> With out this line i got grinding. Likely a later move relies on destination.

Hmm. I would be happier if we knew why we were doing things. The state of `destination` and all the XYZ positioning is, after all, totally [deterministic](https://en.wikipedia.org/wiki/Deterministic_algorithm).

**Comment by thinkyhead** [comment]
I'd like to know if any of these changes affect #4206.

Related to that, do you think it's possible that someplace we're now calling `sync_plan_position()` where we should be using `SYNC_PLAN_POSITION_KINEMATIC()`? I'm on the verge of going back and comparing line-by-line the leveling code from 2 weeks ago to today's to see if I can stumble on the important difference.

**Comment by thinkyhead** [comment]
Merging… with caveats. I'll soon add more detailed logging to track all changes to `destination` and `current_position` and all calls to functions that either alter them or send them to the planner.

**Pull request merged**